### PR TITLE
Pictures collection - WIP

### PR DESCRIPTION
**Pull request by remivar**
Added View Trees button in the Home page, added UIcollectionview in the View Trees module.  
path: TreeTracker - UI - Views - View Trees

### CODE REVIEW

File: TreeTracker.xcodeproj/project.pbxproj
Comment (AlexCornforth) [inline TreeTracker.xcodeproj/project.pbxproj:993]: Lets not check in development team

File: TreeTracker.xcodeproj/project.pbxproj
Comment (AlexCornforth) [inline TreeTracker.xcodeproj/project.pbxproj]: Lets not check in development team

File: TreeTracker/UI/Views/AddTree/AddTreeViewController.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/AddTree/AddTreeViewController.swift]: Get why you did this for dev purposes (the main branch now has a photo library option for debug builds to resolve this) but they need to be uncommented before merging.

File: TreeTracker/UI/Views/Authentication/Selfie/SelfieViewController.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/Authentication/Selfie/SelfieViewController.swift]: Get why you did this for dev purposes (the main branch now has a photo library option for debug builds to resolve this) but they need to be uncommented before merging.

File: TreeTracker/UI/Views/Home/Home.storyboard
Comment (AlexCornforth) [inline TreeTracker/UI/Views/Home/Home.storyboard:43]: I'd prefer to keep setting of colours in code - it reduces chance of conflicts and is much more human readable than huge storyboard xml files.

File: TreeTracker/UI/Views/Home/HomeViewController.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/Home/HomeViewController.swift]: Can we remove this print statement - feel free to use the Logger in the main branch though if you want to log something to the console in debug only.

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: Can we remove this commented out line if its not used.

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: Maybe rename to something like 'viewTreesCollectionview` so its clear what it is

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: This still needs to be populated via the viewmodel im guessing?

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: You can use the AlertPresenting protocol here to easily present an alert. Check the other view controllers for examples

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: We should reload the collection view here

File: TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift]: We should delegate back to the view here with updated data so it can reload the collection view

**Review by AlexCornforth — CHANGES_REQUESTED** [review]
Nice work @remivar - like that you following the existing patterns we have in place. I think it would make sense to pull through the actual data to the view controller though as part of this PR.

File: TreeTracker/UI/Views/AddTree/AddTreeViewController.swift
Comment (remivar) [reply TreeTracker/UI/Views/AddTree/AddTreeViewController.swift]: my  fault, that was for testing the simulator

File: TreeTracker/UI/Views/Authentication/Selfie/SelfieViewController.swift
Comment (remivar) [reply TreeTracker/UI/Views/Authentication/Selfie/SelfieViewController.swift]: Ditto

File: TreeTracker/UI/Views/Home/Home.storyboard
Comment (remivar) [reply TreeTracker/UI/Views/Home/Home.storyboard:43]: gotcha...

File: TreeTracker/UI/Views/Home/HomeViewController.swift
Comment (remivar) [reply TreeTracker/UI/Views/Home/HomeViewController.swift]: removed

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: I will fix

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: I will update

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: ok

File: TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift]: Are we using `TreeCountData` anywhere in the ViewModel? We should remove it if we don't need it.

File: TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift:35]: We should pass the error to view here:

`viewDelegate?.viewTreesViewModel(self, didReceiveError: error)`

File: TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift]: Can we remove the print. Or use the Logger.swift (its in the master branch - you may need to rebase/merge with master to access)

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: We should present the error alert here:
`present(alert: .error(error))`

File: Podfile
Comment (AlexCornforth) [inline Podfile]: I can't see this used anywhere? We should probably discuss if we want to include this dependency or not. My personal preference would be not to.

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: So looks like we are fetching data on the main thread here. This will cause the UI to freeze up. We should load the image inside the cell, we can then also make use of cell reuse and cancel any ongoing download tasks. Otherwise we could end up with incorrect images on cells.

Also there is a lot of force unwrapping  - I think SwiftLint will throw a warning about this. We shouldn't force unwrap as it can cause a crash, we should handle the missing data more gracefully.

Happy to help you with this if you wish.

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (AlexCornforth) [inline TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: We can remove this function if we don't need it.

**Review by AlexCornforth — CHANGES_REQUESTED** [review]
Looks like there a quite a few conflicts to be resolved before this can be merged, happy to help you with that if needed.

We need to resolve the fetching of images in the cells, currently it will cause issues.

Lets also discuss about if we need to include SDWebImage or not.

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: I have made the changes in ViewTreesViewController and ViewTreesViewModel

File: TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift
Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift]: I did remove that print statement

File: Podfile
Comment (remivar) [reply Podfile]: I used it earlier, but later changed. Will remove it

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: we may want to use DidSelectItem func later on.

File: TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift
Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift]: okay

File: TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift
Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewModel.swift:35]: I will fix that

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: will create the alert controller

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: I will fix that

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: I removed

File: TreeTracker.xcodeproj/project.pbxproj
Comment (remivar) [reply TreeTracker.xcodeproj/project.pbxproj:993]: I changed the team name

File: TreeTracker.xcodeproj/project.pbxproj
Comment (remivar) [reply TreeTracker.xcodeproj/project.pbxproj]: I changed team name

File: TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift
Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: I made some changes in the code to fetch the pics without  using main thread. However it didn't work out. You can fix this and please let me know.

Comment (AlexCornforth) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: Here is a tutorial you can use to learn about this: https://cocoacasts.com/fm-3-download-an-image-from-a-url-in-swift

Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: you can help me on this.

Comment (remivar) [reply TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift]: hope we have fixed this issue

**Comment by remivar** [comment]
closing the pr as it has lots of conflicts, created another one from branch feature/17/pictures collection

**Pull request closed without merging**
